pycontext/context.py

```diff
@@ -109,6 +110,25 @@ def __getitem__(self, key):
             raise KeyError(key)
         return value
 
+    def __getattr__(self, key):
+        """
+        Get attribute's value, starting at the current scope and going upward.
+
+        :param key: the name of the attribute
+        :param value: the variable value
+        """
+        if key in object.__getattribute__(self, '__slots__'):
+            super(Context, self).__getattr__(key)
+        else:
+            value, frame = self._find(key)
+            if frame is None:
+                raise AttributeError(
+                    "'{cls}' instance has no attribute '{name}'".format(
```

Comment:
hmm. manually formatting error message might cause the error message to not match in some python versions. how about something like:

```python
try:
    return super(Context, self).__getattr__(key)
except AttributeError:
    value, frame = self._find(key)
    if frame is not None:
        return value
    raise
```

since `__getattr__` is only called when attribute is already not found in the class, this should not add too much overhead so no need to explicitly check for `__slots__`